### DISCUSSION TITLE
c-store commands under single association request

### DISCUSSION THREAD

**Discussion by liorkats** [fo-dicom/fo-dicom#1409] [Q&A] [answered]
Hello ,

Is it possible to make several c-store commands under single association request ?
example will be appreciated.

best regards
Lior

**Comment by gofal** [comment, marked as answer]
it's as simple as

```
var client = DicomClientFactory.Create("127.0.0.1", 12345, false, "SCU", "ANY-SCP");

await client.AddRequestAsync(new DicomCStoreRequest(@"test1.dcm"));
await client.AddRequestAsync(new DicomCStoreRequest(@"test2.dcm"));
// here you can add as many c-store commands as you want

// now before sending is injected, the various presentation contexts required for all the reuqests are calcualted
await client.SendAsync();
```

**Reply by liorkats** [reply]
@gofal 
Thank you for your answer.
In this case a add as much files as I want and they will be sent in one association.
How should I handle the response status ?

**Reply by gofal** [reply]
each request has an event "OnResponceReceived". so in this small sample the new created Requests are added to the client. if you want to track the response, then you would have to do something like

```
var request = new DicomCStoreRequest(@"test1.dcm"));
request.OnResponseReceived = (request, response) => Console.WriteLine("test1.dcm: " + response.Status.ToString());
await client.AddReqeustAsync(request);

request = new DicomCStoreRequest("....");
....
```
but obviously you would programm it in a smarter way with some loops and collections.
